lib/schema.js

```diff
@@ -13,6 +12,29 @@ const addSchema = Symbol('addSchema');
 const factorify = Symbol('factorify');
 const initActions = Symbol('initActions');
 const validate = Symbol('validate');
+const classifyDomains = Symbol('classifyDomains');
+
+const domainConstructors = { Enum, Flags };
+const classWrappers = {
+  Uint8Array: value => Uint8Array.from(value),
+  HashMap: value => {
+    try {
+      const map = new Map(value);
+      return map;
```

Comment:
I think that this should be `null`, not an empty `Map` because value cannot be converted to a Map.

Reply:
IFAIU, `try {} catch` is in case someone attempts to pass a value that cannot be converted to `Map` in that case constructor will throw but we need to return `null`, no raise an Error.

Reply:
In case your point is: "You can combine this lith with the previous one into: `return new Map(value)`" - you have +1 for that.